ldk/go/exceptionLoggingInterceptor.go

```diff
@@ -0,0 +1,53 @@
+package ldk
+
+import (
+	"context"
+	"regexp"
+
+	"google.golang.org/grpc"
+)
+
+type ExceptionLoggingInterceptor struct {
+	logger          *Logger
+	interceptedConn *grpc.ClientConn
+}
+
+func (eli *ExceptionLoggingInterceptor) Invoke(ctx context.Context, method string, args interface{}, reply interface{}, opts ...grpc.CallOption) error {
+	err := eli.interceptedConn.Invoke(ctx, method, args, reply, opts...)
+	if err != nil {
+		eli.logError(method, err)
+	}
+
+	return err
+}
+
+func (eli *ExceptionLoggingInterceptor) NewStream(ctx context.Context, desc *grpc.StreamDesc, method string, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+	stream, err := eli.interceptedConn.NewStream(ctx, desc, method, opts...)
+	if err != nil {
+		eli.logError(method, err)
```

Comment:
no clue how gofmt was okay with this :/

Reply:
What's wrong with it. I must be missing something.

Reply:
Oh yeah, it's indenting really far, but then I realized GoLang uses tabs. My bad